### DISCUSSION TITLE
Custom build fails to use the system h264 encoder

### DISCUSSION THREAD

**Discussion by hashworks** [OvenMediaLabs/OvenMediaEngine#1150] [Help]
I'm the maintainer of the [Arch Linux OME package](https://github.com/archlinux/svntogit-community/blob/packages/ovenmediaengine/trunk/PKGBUILD#L11), which is build to use the system libraries (`/usr/lib`) and does not use the `prerequisites.sh` script.

It works fine with a bypass profile, but starting a stream that requires h264 encoding fails with `E [Dech264:2251236] Transcoder | encoder_avc_openh264.cpp:131  | Could not find encoder: 27 (h264)` (which references [this line](https://github.com/AirenSoft/OvenMediaEngine/blob/v0.15.6/src/projects/transcoder/codec/encoder/encoder_avc_openh264.cpp#L131)).

One could think that it can't find the `libx264.so` library, but `ldd /usr/bin/OvenMediaEngine` [says otherwise](https://fb.hash.works/KYZet/):
```
[...]
	libx264.so.164 => /usr/lib/libx264.so.164 (0x00007f5997600000)
[...]
```

[This is the config I'm using](https://fb.hash.works/KjknB5n/). Sending a stream 'foo' to the RTMP port successfully starts the `foo` OutputStream, but fails to create the `foo_encode` OutputStream with the aforementioned error.

Any ideas?

**Comment by hashworks** [comment]
My bad: This error happens since for encoding with h264 OvenMediaEngine requires ffmpeg to be build with `--enable-libopenh264`, which isn't enabled in the Arch ffmpeg package.

**Reply by getroot** [reply]
Yes, due to license conflict (OvenMediaEngine:AGPLv3 / x264: GPLv2), OvenMediaEngine removed x264 a long time ago and are using OpenH264.
